portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,13 +27,30 @@
 public class DataServlet extends HttpServlet {
   private final List <String> messages = new ArrayList<>();
   private final Gson gson = new Gson();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    messages.add("Something crazy");
-    messages.add("Something simple");
-    messages.add("Something amazing");
     String json = gson.toJson(messages);
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String username = getParameter(request , "username", "Anonymus");
+    String message = getParameter(request , "message-data","");
+    messages.add(username +": "+ message);
+    int indexOfRecent = messages.indexOf("" + username +": "+ message + "");
```

Comment:
I would suggest to store the String in a variable, and add it to messages, then print it. This is more direct than adding the string, then retrieving its index.